### ISSUE TITLE
PaymentMethod, Buyer, etc. (foreign keys) are also navigation properties in Order but they are not child entities

### ISSUE THREAD

**Issue by CESARDELATORRE** [dotnet/eShopOnContainers#16]
Related to: https://github.com/dotnet/eShopOnContainers/issues/18

The Enumerable object PaymentMethod is child of the  Aggregate-Roots "Buyer". Ok, that's right.

However, in the Order AGGREGATE-ROOT, because it is referenced through the EF graph (navigation property), it feels like it is also a child-VO (embedded). Can we fix that? 
Do we really need it as an object within the Order entity?

If we maintain it as a reference through the EF graph..., how can we make clearer in the model that it is NOT a child entity but just a reference?

**This is RIGHT because PaymentMethod is a child of Buyer**

```
    public class Buyer
       :Entity,IAggregateRoot
    {
        public string FullName { get; private set; }
        public HashSet<PaymentMethod> PaymentMethods { get; private set; }
```

**The following "feels" wrong because Buyer, PaymentMethod are NOT Order's childs **

```
    public class Order
        : Entity, IAggregateRoot
    {
        public int BuyerId { get; private set; }
        <b>public Buyer Buyer { get; private set; } </b>
        public DateTime OrderDate { get; private set; }
        //...
        public int PaymentMethodId { get; private set; }
        <b>public PaymentMethod PaymentMethod { get; private set; } </b>
```

I know that in the ConfigureOrder(), the PaymentMethod is configured as:

            orderConfiguration.HasOne(o => o.PaymentMethod)
                .WithMany()
                .HasForeignKey(o => o.PaymentMethodId)
                .OnDelete(DeleteBehavior.Restrict);

So, as a ForeignKey, but, do we really need the "object" here referenced through the EF graph? 
Can we have just the PaymentMethodID in the Order so it is clearer that the PaymentMethod is NOT a child-entity of the Order Aggregate-Root?

Please, review.

**Comment by eiximenis** [comment]
That's solved in DEV branch (commits a7c7426659ac8dfa8791edfce26ee20eeede3e9c & be440a9a02b53f60021147645a13843cf9011ce6)
I left the migrations: they're empty (which is good because DB schema is the same) but the ModelSnapshot is slightly different, so I left them for safety.

**Comment by CESARDELATORRE** [comment]
I'll close this issue when we merge to MASTER.
